tests/yast2_gui/yast2_firewall.pm

```diff
@@ -68,41 +69,129 @@ sub susefirewall2 {
-    assert_screen 'firewall-ipsets-entry';
-
-    # now close and exit
-    assert_and_click 'firewall-file-menu';
-    assert_and_click 'firewall-file-quit';
+sub stop_service {
+    assert_screen 'yast2_firewall_start-up';
+    assert_screen 'yast2_firewall_service_status_running';
+    send_key $fw{service_stop};
+    assert_screen [qw(yast2_firewall_service_status_stopped generic-desktop)];
+    if (match_has_tag('generic-desktop')) {
+        record_soft_failure "bsc#1114677 - Dialog dissapear after switching service status";
+        assert_screen 'yast2_firewall_service_status_stopped';
+    }
+}
+
+sub start_service {
+    assert_screen 'yast2_firewall_start-up';
+    assert_screen 'yast2_firewall_service_status_stopped';
+    send_key $fw{service_start};
+    assert_screen [qw(yast2_firewall_service_status_running generic-desktop)];
+    if (match_has_tag('generic-desktop')) {
+        record_soft_failure "bsc#1114677 - Dialog dissapear after switching service status";
+        assert_screen 'yast2_firewall_service_status_running';
+    }
+}
+
+sub verify_interface {
+    my (%args) = @_;
+
+    assert_and_click 'yast2_firewall_interfaces_menu';
+    assert_screen 'yast2_firewall_interfaces_' . $args{device} . '_' . $args{zone};
+}
+
+sub change_interface_zone {
+    my $zone = shift;
+
+    assert_and_click 'yast2_firewall_interfaces_menu';
+    assert_screen 'yast2_firewall_interfaces';
+    send_key $fw{interfaces_change_zone};
+    assert_screen 'yast2_firewall_interfaces_change_zone';
+    send_key $fw{interfaces_change_zone_zone};
+    type_string "$zone\n";
+}
+
+sub verify_zone {
+    my (%args) = @_;
+
+    my $interfaces = $args{interfaces} //= 'no_interfaces';
+    my $default    = $args{default}    //= 'no_default';
+
+    assert_and_click 'yast2_firewall_zones';
+    assert_screen 'yast2_firewall_' . $args{name} . '_' . $interfaces . '_' . $default;
+}
+
+sub set_default_zone {
+    my $zone = shift;
+
+    assert_and_click 'yast2_firewall_zone_' . $zone;
+    send_key $fw{zones_set_as_default};
+}
+
+sub configure_zone {
+    my (%args) = @_;
+
+    assert_and_click 'yast2_firewall_zones';
+    if ($args{service}) {
+        assert_and_click 'yast2_firewall_zone_' . $args{zone} . '_menu';
+        assert_and_click 'yast2_firewall_zone_service_known_scroll_on_top';    # assuming allowed list empty
+        send_key_until_needlematch 'yast2_firewall_zone_service_' . $args{service} . '_selected', 'down';
+        send_key $fw{zones_service_add};
+        assert_screen 'yast2_firewall_zone_service_' . $args{service} . '_allowed';
+    }
+    if ($args{port}) {
+        send_key $fw{zones_ports};
+        assert_screen 'yast2_firewall_zone_service_warning';
+        send_key $fw{yes};
+        assert_screen 'yast2_firewall_zone_ports_tab_selected';
+        send_key $fw{tcp};
+        type_string '7777';
+    }
+}
+
+sub configure_firewalld {
+    record_info('Start-Up', "Managing the firewalld service");
+    stop_service;
```

Comment:
What's about service status asserting and not just believing that GUI did all the work?

Reply:
good suggestion, I added a new bug for that in the first comment along with VR updated.

Reply:
to verify right after the service I have to come back to the user console because I was ask for user password in the UI, very strange..., and now a "workaround box" appears. Do you think is correct the transition?